l1-contracts/src/core/libraries/rollup/ProposeLib.sol

```diff
@@ -62,6 +62,49 @@ struct ValidateHeaderArgs {
   BlockHeaderValidationFlags flags;
 }
 
+/**
+ * @title ProposeLib
+ * @author Aztec Labs
+ * @notice Library responsible for handling the L2 block proposal flow in the Aztec rollup.
+ *
+ * @dev This library implements the core block proposal mechanism that allows designated proposers to submit
+ *      new L2 blocks to extend the rollup chain. It orchestrates the entire proposal process including:
+ *      - Blob validation and commitment calculation
+ *      - Header validation against chain state and timing constraints
+ *      - Validator selection and proposer verification
+ *      - Fee calculation and mana consumption tracking
+ *      - State transitions and archive updates
+ *      - Message processing between L1 and L2 via the Inbox and Outbox contracts
+ *
+ *      The proposal flow operates within Aztec's time-based model where:
+ *      - Each slot has a designated proposer selected from the validator set
+ *      - Blocks must be proposed in the correct time slot and build on the current chain tip
+ *      - Proposers must provide valid attestations from committee members
+ *      - All state transitions are atomically applied upon successful validation
+ *
+ *      Key functions:
+ *      - `propose`: Main entry point called from `RollupCore.propose`.
+ *         Handles the complete block proposal process from validation to state updates.
+ *      - `validateHeader`: Validates block header against chain state, timing, and fee requirements.
+ *         Called internally from `propose`, and externally from `RollupCore.validateHeaderWithAttestations`,
+ *         used by proposers to ensure the header is valid before submitting the tx.
+ *
+ *      Dependencies on other main libraries:
+ *      - STFLib: State Transition Function library for chain state management, pruning, and storage access
+ *      - FeeLib: Fee calculation library for mana pricing, L1 gas oracles, and fee header computation
+ *      - ValidatorSelectionLib: Validator and committee management for epoch setup and proposer verification
+ *      - BlobLib: Blob commitment validation and hash calculation for data availability
+ *      - ProposedHeaderLib: Block header hashing and validation utilities
+ *
+ *      Security considerations:
+ *      - Only the designated proposer for the current slot can propose a block, enforced by
+ *        validating the proposer validator signature among attestations. All other attestations are not
+ *        verified on chain until time of proof submission.
+ *      - Each block must built on the immediate previous one, ensuring no forks. This is enforced by checking
+ *        the last archive root and block numbers. If the previous block is invalid, the proposer is expected to
+ *        first invalidate it.
+ *      - Blob commitments are validated.
```

Comment:
It is not super clear what this actually means, could be understood as the point evaluation.